test/Microsoft.Framework.Caching.Distributed.Tests/CacheServiceExtensionsTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Framework.Caching.Memory;
+using Microsoft.Framework.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.Framework.Caching.Distributed
+{
+    public class CacheServiceExtensionsTests
+    {
+        [Fact]
+        public void AddCaching_RegistersMemoryCacheAsSingleton()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+
+            // Act
+            services.AddCaching();
+
+            // Assert
+            var memoryCache = services.Where(desc => desc.ServiceType == typeof(IMemoryCache)).FirstOrDefault();
+
+            Assert.NotNull(memoryCache);
+            Assert.Equal(ServiceLifetime.Singleton, memoryCache.Lifetime);
+        }
+
+        [Fact]
+        public void AddCaching_RegistersDistributedCacheAsTransient()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+
+            // Act
+            services.AddCaching();
+
+            // Assert
+            var distributedCache = services.Where(desc => desc.ServiceType == typeof(IDistributedCache))
```

Comment:
nit: I think you can move the predicate directly into FirstorDefault eliminating the need for the intermediate Where()